### ISSUE TITLE
inference_type "clf" does not work

### ISSUE THREAD

**Issue by abedini-arteriaai** [IBM/fastfit#9]
Hi all, FastFit is great - thank you for sharing. 

I wanted to point out the classification inference_type does not work. 

the code below will error out with `FastFitTrainable.clf_loss() got multiple values for argument 'return_scores'`. I'm assuming it's because `clf_loss` only takes encodings, attention_mask and labels (3, not 4). 

```
elif self.config.inference_type == "clf":
                _, scores = self.clf_loss(
                    query_encodings,
                    doc_encodings,
                    query_attention_mask,
                    doc_attention_mask,
                    return_scores=True,
                )
```

https://github.com/IBM/fastfit/blob/a9a117753e0d6ff5b9a98848e6b6673798c253d7/fastfit/modeling.py#L490C1-L499C1

**Comment by elronbandel** [comment]
I added a fix from different version of this code. Can you try again? otherwise i will add later on a dedicated test and re solve

**Comment by elronbandel** [comment]
Fixed
